API/WithDocs/Microsoft.Extensions.Http.AutoClient/DeleteAttribute.cs

```diff
@@ -0,0 +1,63 @@
+// Assembly 'Microsoft.Extensions.Http.AutoClient'
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Extensions.Http.AutoClient;
+
+/// <summary>
+/// Defines an API <c>DELETE</c> method.
+/// </summary>
+/// <remarks>
+/// Marks a method within an interface annotated with <see cref="T:Microsoft.Extensions.Http.AutoClient.AutoClientAttribute" /> as an API <c>DELETE</c> method.
+///
+/// The return type of an API method must be a <c>Task&lt;T&gt;</c>.
+/// If T is a <see cref="T:System.String" /> and the dependency returns "text/plain" content type, the result will be the raw content of the response. Otherwise, it will be deserialized from JSON.
+/// If T is of type <see cref="T:System.Net.Http.HttpResponseMessage" />, the result will be the actual response message without further processing.
+///
+/// If you provide an extra parameter to the method, you should use it between curly braces in the URL to make it a URL parameter. For example: <c>/api/users/{userId}</c>.
+/// </remarks>
+/// <example>
+/// <code>
+/// [AutoClient("MyClient")]
+/// interface IMyDependencyClient
+/// {
+///     [Delete("/api/users/{userId}")]
+///     Task&lt;bool&gt; DeleteUserAsync(string userId, CancellationToken cancellationToken = default);
+/// }
+/// </code>
```

Comment:
it might  be helpful for these attributes to mention describe what it means to apply them without parameters.

Reply:
I don't think you can, looks like all of these attributes have a constructor that requires a path. Even if the API you want to get to is in the root of the base URL, you'd write something like:

```c#
[Delete("/")]
Task<bool> DeleteAllUsersAsync(CancellationToken ct);
```

Reply:
Essentially the source generator requires a path to be set in order to fill the implementation of the client.

Reply:
I didn't see any validation that the path is not empty, nor tests for that case

Reply:
@danmoseley What validation should we be doing? I believe throwing ArgumentNullException in an attribute definition is frowned upon. Do you want us to test that the code generator produces an error if the user passes a null! ?

@dbarbosapn

Reply:
for a test yes the test would be of the code that read the values off it (maybe that test exists)

my question was whether we should document what empty string does, whether it's useful or not. It's probably obvious that null is an error (you'd have to write `[Delete(null!)]` after all, as you say)

my care level is low, just pointing out it may not be obvious.

Reply:
Ah, "" rather than null.

@dbarbosapn Can you add some text in the docs explaining what the design intent is there, what will happen with ""? And I checked, and it's possible that the code generator would crash if someone passed null! in the constructor to the attributes. So I think that's worth fixing too, the generator should never crash even on bogus input.

Reply:
BTW I see some attribute constructors in the core libraries that could potentially throw, but as you say, it's probably not a good pattern.

Reply:
I'll take a look and write tests for this as soon as I can. It was for sure an oversight. The initial design idea was to rely as much as possible on the analyzer part of the generator to provide as detailed insights as possible during REST client development, through the IDE's diagnostics, not to throw exceptions for sure.

Reply:
Tracked by https://github.com/dotnet/extensions/issues/4463